src/main/java/org/folio/rest/support/ShelvingOrderUpdate.java

```diff
@@ -0,0 +1,186 @@
+      .append("jsonb AS \"itemJson\" ")
+      .append("FROM item WHERE not (jsonb ? 'effectiveShelvingOrder') ")
+      .append(" FOR UPDATE ")
+      .append("LIMIT $1").toString();
+
+
+  private static final String SQL_UPDATE_ITEMS = new StringBuilder()
+      .append("UPDATE ITEM ")
+      .append("SET jsonb = $1 ")
+      .append("WHERE id = $2").toString();
+
+  // Defines version threshold where this property was introduces from
+  private static final String FROM_VERSION_SHELVING_ORDER = "19.5.0";
+
+  private static final String OPERATION_STATUS_VERSION_MISMATCH = "Module version isn't expected";
+  private static final String OPERATION_STATUS_ITEMS_UPDATES_SUCCESSFUL = "Items updates completed successfully";
+  private static final String OPERATION_STATUS_ITEMS_UPDATES_FAILED = "Items updates failed";
+
+
+  private int updateBunchSize;
+  private Map<UUID, String> itemsShelvingOrderMap;
+  private Map<UUID, Item> itemsMap;
+
+  public ShelvingOrderUpdate(int updateBunchSize) {
+    super();
+    this.updateBunchSize = updateBunchSize;
+    setFromModuleVersion(FROM_VERSION_SHELVING_ORDER);
+    itemsShelvingOrderMap = new LinkedHashMap<>();
+    itemsMap = new LinkedHashMap<>();
+  }
+
+  public Future<String> updateItems(TenantAttributes attributes, Map<String, String> headers, Context vertxContext) {
+    Promise<String> promise = Promise.promise();
+
+    String fromModuleVersion = attributes.getModuleFrom().trim();
+    if (StringUtils.isBlank(fromModuleVersion) || isNewForThisInstall(fromModuleVersion)) {
+      // Update items only for versions before the expected
+      promise.fail(OPERATION_STATUS_VERSION_MISMATCH);
+    } else {
+      PostgresClient postgresClient =  PgUtil.postgresClient(vertxContext, headers);
+
+      hasItemsToUpdate(vertxContext.owner(), postgresClient)
+          .onSuccess(itemsExists -> {
+            log.warn("Items update started");
+          }).onFailure(h -> {
+            log.info("No items for update");
+      });
+    }
+
+    return promise.future();
+  }
+
+  private Future<Boolean> hasItemsToUpdate(Vertx vertx, PostgresClient postgresClient) {
+    Promise<Boolean> promise = Promise.promise();
+
+    postgresClient.execute(SQL_EXISTS_ITEMS_TO_UPDATE, ar -> {
+      if (ar.succeeded()) {
+        RowSet<Row> rows = ar.result();
+        boolean itemsExists = rows.iterator().next().getBoolean(0);
+        promise.complete(itemsExists);
+
+        fetchAndUpdatesItems(vertx, postgresClient)
+            .onSuccess(h -> hasItemsToUpdate(vertx, postgresClient))
+            .onFailure(h -> log.warn("Current items updates failed: "  + h.getMessage()));
+
+      } else {
+        promise.fail("Failure: " + ar.cause().getMessage());
+      }
+    });
+
+    return promise.future();
+  }
+
+  private Future<Integer> fetchAndUpdatesItems(Vertx vertx, PostgresClient postgresClient) {
+    Promise<Integer> promise = Promise.promise();
+
+    vertx.setTimer(500L, timer -> {
```

Comment:
Why is a timer being used for these updates?

I think a timer introduces some additional challenges for this process. It means that multiple batches might be processed at the same time, which I think is counter-productive to the intent of using batches to maintain a consistent load. Alternatively, if each batch takes less than 500ms, it might mean the process is slower than it needs to be.

Reply:
Using of timer here was attempt to not load main verticle with log-time update operation. Do you mean that running of blocking code approach will be better here?

Reply:
> Using of timer here was attempt to not load main verticle with log-time update operation. Do you mean that running of blocking code approach will be better here?

I am not advocating for blocking code.

If the intent is to repeatedly run the code for a batch until there are no longer any items left to update, I suggest triggering the next batch when the last batch finishes. 

One approach is to trigger the batches recursively, e.g. https://github.com/folio-org/mod-circulation/blob/494e73fab76f637bc6c35603d3669e6214206cd4/src/main/java/org/folio/circulation/support/fetching/PageableFetcher.java#L37

Reply:
>I suggest triggering the next batch when the last batch finishes

Reply:
You can run for half an hour if you want. The async tenant API can handle it. Avoid cleverness like this.

Reply:
Do you mean that it's need to avoid of using "setTimer" and just execute even huge amount  items update via vertx postgres client directly, even without leveraging of "execute blocking"?

Reply:
@oleksandr-yatsenko1 

> I'll take a look the code by link, looks like that's background fetching of data for some kind of UI consumer like a navigable table.

The purpose of that code is different. It only fetches records. The principle is the same I believe, at the end of each page / batch of records, the next is fetched. The primary difference I can think of is that this process needs to also update records.

Reply:
If I understand correctly the better approach will be to use some kind of working pool to run several update processes simultaneously. Because Postgres Client possibly do the same under the hood, i.e. use typical executeBlocking like vertx JDBC does. That why possibly we don't need to care about blocking of main event-loop by database update operation. The only difference I think could be in multi-processing, it will be faster of course but require extra tracking of updating rows (possibly by remember borders of rowids or items ids), to avoid lock waits until parallel update completed (if any of rows range will cross, without such range control). Currently  there is only single update process supposed, possibly it will be enough for this, especially takin into account that this should be performed only once. But in general,  I think we lack of such functionality of data updates because unify approach, possibly with multi-processing capabilities should be introduced to keep data in consistency.